### PR TITLE
documentation reconstructors

### PR DESCRIPTION
**Pull request by Zaineb18**

### CODE REVIEW

**Comment by chaithyagr** [comment]
@Zaineb18 can you please handle the conflicts and maybe rebase your changes?

**Comment by paquiteau** [comment]
I fixed the conflicting merge. Ready for merging ?

**Comment by chaithyagr** [comment]
ahh no, need to review :) Adding you to review @paquiteau

File: mri/reconstructors/base.py [hunk 1]
```diff
@@ -39,26 +39,28 @@ class ReconstructorBase(object):
...
         For the Synthesis case, finds the solution of:
         ..math:: (1/2) * ||F Wt alpha - y||^2_2 + mu * H(alpha)
+        with ..math:: alpha = W x and x = Wt alpha
 
     Parameters
     ----------
-    fourier_op: instance of OperatorBase
-        Defines the fourier operator F.
+    fourier_op: instance of FourierOperatorBase
+        Defines the fourier operator F in the above equation
```
Comment (paquiteau) [inline mri/reconstructors/base.py:47]:
```suggestion
        Defines the fourier operator :math:`F` in the above equation
```

File: mri/reconstructors/base.py [hunk 2]
```diff
@@ -39,26 +39,28 @@ class ReconstructorBase(object):
...
     linear_op: object
-        Defines the linear sparsifying operator W. This must operate on x and
+        Defines the linear sparsifying operator denoted W in the above equation. This must operate on x and
```
Comment (paquiteau) [inline mri/reconstructors/base.py:49]:
```suggestion
        Defines the linear sparsifying operator denoted :math:`W` in the above equation. This must operate on x and
```

File: mri/reconstructors/base.py [hunk 3]
```diff
@@ -39,26 +39,28 @@ class ReconstructorBase(object):
...
         have 2 functions, op(x) and adj_op(coeff) which implements the
         operator and adjoint operator. For wavelets, this can be object of
-        class WaveletN or WaveletUD2 from mri.operators
+        class WaveletN or WaveletUD2 from mri.operators.linear
```
Comment (paquiteau) [inline mri/reconstructors/base.py:52]:
```suggestion
        class WaveletN or WaveletUD2 from `mri.operators.linear`
```

File: mri/reconstructors/base.py [hunk 4]
```diff
@@ -39,26 +39,28 @@ class ReconstructorBase(object):
...
     regularizer_op: operator, (optional default None)
         Defines the regularization operator for the regularization function H.
         If None, the  regularization chosen is Identity and the optimization
-        turns to gradient descent.
+        turns to gradient descent. Defines H in the above equation.
```
Comment (paquiteau) [inline mri/reconstructors/base.py:56]:
```suggestion
        turns to gradient descent. Defines :math:`H` in the above equation.
```

File: mri/reconstructors/base.py [hunk 5]
```diff
@@ -39,26 +39,28 @@ class ReconstructorBase(object):
...
     gradient_formulation: str between 'analysis' or 'synthesis',
         default 'synthesis'
         defines the formulation of the image model which defines the gradient.
-    grad_class: Gradient class from mri.operators.
+    grad_class: Gradient class from mri.operators.gradient.
```
Comment (paquiteau) [inline mri/reconstructors/base.py:60]:
```suggestion
    grad_class: Gradient class from `mri.operators.gradient`.
```

File: mri/reconstructors/base.py [hunk 6]
```diff
@@ -39,26 +39,28 @@ class ReconstructorBase(object):
...
         Points to the gradient class based on the MR Image model and
-        gradient_formulation.
+        gradient_formulation. Can of be of class GradAnalysis, GradSynthesis,
+        GradSelfCalibrationAnalysis or GradSelfCalibrationSynthesis
```
Comment (paquiteau) [inline mri/reconstructors/base.py:63]:
```suggestion
        gradient_formulation. 
```
Move to See Also Section

File: mri/reconstructors/calibrationless.py [hunk 7]
```diff
@@ -29,15 +29,17 @@ class CalibrationlessReconstructor(ReconstructorBase):
...
     ----------
     fourier_op: instance of OperatorBase.
         Defines the fourier operator F in the above equation.
-    linear_op: instance of OperatorBase, default None.
-        Defines the linear sparsifying operator W. This must operate on x and
-        have 2 functions, op(x) and adj_op(coeff) which implements the
-        operator and adjoint operator. For wavelets, this can be object of
-        class WaveletN or WaveletUD2 from mri.operators .
+    linear_op: OperatorBase, default None
+        Defines the linear sparsifying operator denoted W in the equation above.
```
Comment (paquiteau) [inline mri/reconstructors/calibrationless.py:39]:
```suggestion
        Defines the linear sparsifying operator denoted :math:`W` in the equation above. 
```

File: mri/reconstructors/self_calibrating.py [hunk 8]
```diff
@@ -45,27 +47,26 @@ class SelfCalibrationReconstructor(ReconstructorBase):
     mri.operators
         Defines the fourier operator F in the above equation.
     linear_op: object, (optional, default None)
-        Defines the linear sparsifying operator W. This must operate on x and
+        Defines the linear sparsifying operator denoted W in the equation above. This must operate on x and
```
Comment (paquiteau) [inline mri/reconstructors/self_calibrating.py:50]:
```suggestion
        Defines the linear sparsifying operator denoted :math:`W` in the equation above. This must operate on x and
```

File: mri/reconstructors/self_calibrating.py [hunk 9]
```diff
@@ -45,27 +47,26 @@ class SelfCalibrationReconstructor(ReconstructorBase):
...
         have 2 functions, op(x) and adj_op(coeff) which implements the
         operator and adjoint operator. For wavelets, this can be object of
-        class WaveletN or WaveletUD2 from mri.operators .
+        class WaveletN or WaveletUD2 from mri.operators.linear .
```
Comment (paquiteau) [inline mri/reconstructors/self_calibrating.py:53]:
```suggestion
        class WaveletN or WaveletUD2 from `mri.operators.linear` .
```

File: mri/reconstructors/self_calibrating.py [hunk 10]
```diff
@@ -45,27 +47,26 @@ class SelfCalibrationReconstructor(ReconstructorBase):
...
         If None, sym8 wavelet with nb_scale=3 is chosen.
     gradient_formulation: str between 'analysis' or 'synthesis',
         default 'synthesis'
         defines the formulation of the image model which defines the gradient.
     kspace_portion: int or tuple (default is 0.1 in all dimension)
-        int or tuple indicating the k-space portion used to estimate the coil
-        sensitivity information.
+        int or tuple indicating the k-space central portion used to estimate the coil
+        sensitivity information, denoted Sl in the above equation.
```
Comment (paquiteau) [inline mri/reconstructors/self_calibrating.py:60]:
```suggestion
        sensitivity information, denoted :math:`Sl` in the above equation.
```

File: mri/reconstructors/single_channel.py [hunk 11]
```diff
@@ -23,19 +23,20 @@ class SingleChannelReconstructor(ReconstructorBase):
...
     ----------
     fourier_op: Instance of OperatorBase.
         Defines the fourier operator F in the above equation.
-    linear_op: Instance of OperatorBase, default None
-        Defines the linear sparsifying operator W. This must operate on x and
-        have 2 functions, op(x) and adj_op(coeff) which implements the
-        operator and adjoint operator. For wavelets, this can be object of
-        class WaveletN or WaveletUD2 from mri.operators .
+    linear_op: OperatorBase, default None
+        Defines the linear sparsifying operator denoted W in the equation above.
```
Comment (paquiteau) [inline mri/reconstructors/single_channel.py:33]:
```suggestion
        Defines the linear sparsifying operator denoted :math:`W` in the equation above. 
```

File: mri/reconstructors/single_channel.py [hunk 12]
```diff
@@ -23,19 +23,20 @@ class SingleChannelReconstructor(ReconstructorBase):
...
-        class WaveletN or WaveletUD2 from mri.operators .
+    linear_op: OperatorBase, default None
+        Defines the linear sparsifying operator denoted W in the equation above. 
+        This must operate on x and have 2 functions, op(x) and adj_op(coeff) 
+        which implements the operator and adjoint operator. For wavelets, this
+        can be object of class WaveletN or WaveletUD2 from `mri.operators.linear`
         If None, sym8 wavelet with nb_scale=3 is chosen.
     regularizer_op: operator, (optional default None)
-        Defines the regularization operator for the regularization function H.
+        Defines the regularization operator for the regularization function denoted H in the equation above.
```
Comment (paquiteau) [inline mri/reconstructors/single_channel.py:39]:
```suggestion
        Defines the regularization operator for the regularization function denoted :math:`H` in the equation above.
```

**Review by paquiteau — COMMENTED** [review]
Lets add some markup where it's required

**Pull request merged**